### ISSUE TITLE
Processing Instructions are ignored

### ISSUE THREAD

**Issue by Arlen22** [nashwaan/xml-js#17]
I have an XML document that includes a processing instruction. But when I run it through xml2js, the processing instruction gets left behind.

```xml
<?xml version="1.0"?>
<?myapi version="13.0"?>
```

**Comment by Arlen22** [comment]
The relevant line is here: https://github.com/nashwaan/xml-js/blob/master/lib/xml2js.js#L82

**Comment by Arlen22** [comment]
And here is a quick modification that I threw together in `xml2js.js`

```js
function onDeclaration(declaration) {
    if (options.ignoreDeclaration) {
        return;
    }
    var key = options.declarationKey;
    var type = "decl";
    if (currentElement[key]) {
        key = options.processingKey;
        currentElement[key] = [];
        type = "proc";
    } else {
        currentElement[key] = {};
    }
    var item = {};
    item[options.nameKey] = declaration.name;
    while (declaration.body) {
        var attribute = declaration.body.match(/([\w:-]+)\s*=\s*"([^"]*)"|'([^']*)'|(\w+)\s*/);
        if (!attribute) {
            break;
        }
        if (!item[options.attributesKey]) {
            item[options.attributesKey] = {};
        }
        item[options.attributesKey][attribute[1]] = attribute[2];
        declaration.body = declaration.body.slice(attribute[0].length); // advance the string
    }
    if (type === "proc") currentElement[key].push(item);
    else currentElement[key] = item;
    if (options.addParent) {
        currentElement[options.declarationKey][options.parentKey] = currentElement;
    }
}
```

**Comment by nashwaan** [comment]
Hi @Arlen22 , thanks for reporting this issue. Can you provide more examples and the expected output, please?

**Comment by nashwaan** [comment]
@Arlen22 
XML specifies that processing instruction [contains just text](http://stackoverflow.com/a/33516121/5619747) not necessarily attributes.

This means both of the following are valid Processing Instructions:
```xml
<?stylesheet some random text?>
```
Output json will be (compact-form):
```json
{"_instruction": {"stylesheet": "some random text"}
```
______
and 
```xml
<?stylesheet href="style.css"?>
```
Output json will be (compact-form):
```json
{"_instruction": {"stylesheet": "href=\"style.css\""}
```
Note that `"_attributes"` will be not generated.

**Comment by Arlen22** [comment]
OK, well I guess that could work. It would be nice if the parser could parse any valid attributes found, and also include the text, since it has the capability built in. Or is that not the intent of this library? Could it be added as a helper method?

Could we do `"stylesheet": { _attributes: {}, _value: "" }`?

But what you wrote above will be enough if you don't want to add this. Because then at least it will not get lost.

**Comment by nashwaan** [comment]
@Arlen22 
```json
"stylesheet": { "_attributes": {}, "_value": "" }}
```
is a good solution, but I am thinking of adding an option like `{instructionHasAttributes: true}` which will parse content of the Processing Instruction as attributes. This mean the output will be:
```json
{"_instruction": {"stylesheet": {"href": "style.css"}}}
```
otherwise if the flag is not set, then we get this:
```json
{"_instruction": {"stylesheet": "href=\"style.css\""}}
```

**Comment by Arlen22** [comment]
That would work as well.

**Comment by nashwaan** [comment]
Published [v1.3.2](https://github.com/nashwaan/xml-js/releases/tag/v1.3.2) which supports processing instruction and `{instructionHasAttributes: true}` flag.

converting this xml:
```xml
<?go to="there"?>
```
will produce in compact mode:
```json
{"_instruction":{"go":"to=\"there\""}}
```
in non-compact (expanded) mode:
```json
{"elements":[{"type":"instruction", "name":"go", "instruction":"to=\"there\""}]}
```
______
and when `{instructionHasAttributes: true}` flag is set, it will produce the following in compact mode:
```json
{"_instruction":{"go":{"_attributes":{"to":"there"}}}}
```
and this in non-compact (expanded) mode:
```json
{"elements":[{"type":"instruction", "name":"go", "attributes":{"to":"there"}}]}
```
_____
@Arlen22 Please let me know if this works for you or not.

**Comment by Arlen22** [comment]
Seems to be working well. Thank you :)
